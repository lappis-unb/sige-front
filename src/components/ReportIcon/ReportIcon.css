.ReportIconCounter {
  font-size: 42px;
  font-weight: bold;
  line-height: 3px;
}

.ReportIconInactive {
  opacity: 0.6;
}

.ReportIconAlignCenter {
  align-content: center;
  justify-content: center;
}
