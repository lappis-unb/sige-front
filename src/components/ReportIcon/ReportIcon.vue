<template>
  <q-card-section
    v-if="reports"
    >
    <div
      class="col text-center ReportIconAlignCenter"
      :class="countReports(reports) === 0 ? 'ReportIconInactive' : ''">
      <p class="ReportIconCounter q-mb-none" id="ReportIconEvents">
        {{ countReports(reports) }}
        <q-icon :name="selectIcon(reports)"/>
      </p>
      <small class="text-center q-mb-none">
        {{this.label}}
      </small>
    </div>
  </q-card-section>
</template>

<script>
export default {
  name: 'ReportIcon',
  props: {
    reports: Object,
    label: String,
    critical: Boolean
  },
  methods: {
    countReports (occurrences) {
      if (this.critical) return (occurrences.critical_tension.length + occurrences.phase_drop.length) || 0
      return (occurrences.precarious_tension.length + occurrences.transductor_connection_fail.length + occurrences.slave_connection_fail.length) || 0
    },
    selectIcon (reports) {
      const path = 'img:/statics/ic_ocorrencia_'
      const iconName = this.critical ? 'critica' : 'precaria'
      return this.countReports(reports) > 0 ? `${path}${iconName}_color.svg` : `${path}${iconName}_mono.svg`
    }
  }
}
</script>

<style>
  @import './ReportIcon.css';
</style>
