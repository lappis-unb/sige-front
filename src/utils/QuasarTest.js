import Vuex from 'vuex'
import Vue from 'vue'
import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import Quasar from 'quasar';

class QuasarTest {
  
  _localVue;

  constructor(){
    this.setUpQuasar()
  }
  
  setUpQuasar() {
    this._localVue = createLocalVue()
    Vue.use(Quasar)
    this._localVue.use(Vuex)
    this._localVue.use(Quasar)
  }

  shallowMount (Component, propsData = {}) {
    const localVue = this._localVue;
    return shallowMount(Component, {
      localVue,
      propsData
    });
  }

  mount (Component, propsData = {}) {
    const localVue = this._localVue;
    return mount(Component, {
      localVue,
      propsData
    });
  }
}

export default new QuasarTest();