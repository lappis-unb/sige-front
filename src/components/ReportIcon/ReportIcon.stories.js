import ReportIcon from '.'

export default {
  title: 'Report Icon',
  component: ReportIcon,
  argTypes: {
    label: { type: 'string' },
    reports: { type: 'object' },
    critical: { type: 'boolean' }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { ReportIcon },
  template: '<report-icon v-bind="$props"/>'
})

export const Default = Template.bind({})
Default.args = {
  label: 'Leves',
  critical: false,
  reports: {
    precarious_tension: [1, 23, 4, 6],
    transductor_connection_fail: [1, 23, 4, 6],
    slave_connection_fail: [1, 23, 4, 6],
    critical_tension: [1],
    phase_drop: [1]
  }
}
