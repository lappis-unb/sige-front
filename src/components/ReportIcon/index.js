import ReportIcon from './ReportIcon.vue';
export default ReportIcon;