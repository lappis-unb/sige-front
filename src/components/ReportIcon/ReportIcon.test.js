import ReportIcon from './ReportIcon.vue'
import QuasarTest from 'src/utils/QuasarTest';

const reportsMocked = {
  precarious_tension: [1, 23, 4, 6],
  transductor_connection_fail: [1, 23, 4, 6 ],
  slave_connection_fail: [1, 23, 4, 6],
  critical_tension: [1],
  phase_drop: [1]
}

describe('Report Icon', () => {
  describe('Teste de snapshot', () => {
    it("Componente precario", () => {
      const wrapper = QuasarTest.shallowMount(ReportIcon, {reports: reportsMocked, label:"Leve", critical: false})
      expect(wrapper.element).toMatchSnapshot()
    })
    it("Componente critico", () => {
      const wrapper = QuasarTest.shallowMount(ReportIcon, {reports: reportsMocked, label:"Leve", critical: true})
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe("Teste do icone", ()=>{
    it('Prop critical deve alterar icone default', () => {
      const wrapper = QuasarTest.shallowMount(ReportIcon, {reports: reportsMocked, label:"Leve", critical: true})
      expect(wrapper.html()).toContain('img:statics/ic_ocorrencia_critica_color.svg')
    })
  })
  describe("Soma dos eventos", ()=>{
    it('O valor de eventos precarios deve ser a soma dos eventos do array', () => {
      const wrapper = QuasarTest.shallowMount(ReportIcon, {reports: reportsMocked, label:"Leve", critical: false})
      const events = Number(wrapper.find('#ReportIconEvents').text())
      expect(events).toBe(12)
    })
    it('O valor de eventos criticos deve ser a soma dos eventos do array', () => {
      const wrapper = QuasarTest.shallowMount(ReportIcon, {reports: reportsMocked, label:"Leve", critical: true})
      const events = Number(wrapper.find('#ReportIconEvents').text())
      expect(events).toBe(2)
    })
  })
})