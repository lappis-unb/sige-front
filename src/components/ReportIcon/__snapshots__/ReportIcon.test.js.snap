// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Icon Teste de snapshot Componente critico 1`] = `
<q-card-section-stub
  tag="div"
>
  <div
    class="col text-center ReportIconAlignCenter"
  >
    <p
      class="ReportIconCounter q-mb-none"
      id="ReportIconEvents"
    >
      
      2
      
      <q-icon-stub
        name="img:statics/ic_ocorrencia_critica_color.svg"
        tag="i"
      />
    </p>
     
    <small
      class="text-center q-mb-none"
    >
      
      Leve
    
    </small>
  </div>
</q-card-section-stub>
`;

exports[`Report Icon Teste de snapshot Componente precario 1`] = `
<q-card-section-stub
  tag="div"
>
  <div
    class="col text-center ReportIconAlignCenter"
  >
    <p
      class="ReportIconCounter q-mb-none"
      id="ReportIconEvents"
    >
      
      12
      
      <q-icon-stub
        name="img:statics/ic_ocorrencia_precaria_color.svg"
        tag="i"
      />
    </p>
     
    <small
      class="text-center q-mb-none"
    >
      
      Leve
    
    </small>
  </div>
</q-card-section-stub>
`;
